--- new_bootstrap.sh	2023-06-20 10:39:32.000000000 -0700
+++ new_bootstrap.patched.sh	2023-06-20 10:42:53.000000000 -0700
@@ -146,7 +146,7 @@
 API_RETRY_ATTEMPTS="${API_RETRY_ATTEMPTS:-3}"
 DOCKER_CONFIG_JSON="${DOCKER_CONFIG_JSON:-}"
 PAUSE_CONTAINER_VERSION="${PAUSE_CONTAINER_VERSION:-3.5}"
-DEFAULT_CONTAINER_RUNTIME="containerd"
+DEFAULT_CONTAINER_RUNTIME="cri-o"
 CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-$DEFAULT_CONTAINER_RUNTIME}"
 IP_FAMILY="${IP_FAMILY:-}"
 SERVICE_IPV6_CIDR="${SERVICE_IPV6_CIDR:-}"
@@ -420,9 +420,21 @@
     systemctl restart docker
     snap set kubelet-eks \
 	container-runtime=docker
+elif [[ "$CONTAINER_RUNTIME" = "cri-o" ]]; then
+  echo "Container runtime is CRI-O"
+  snap set kubelet-eks \
+      container-runtime=remote \
+      container-runtime-endpoint=unix:///var/run/crio/crio.sock
+  dasel put \
+    string \
+    --parser toml \
+    --file /etc/crio/crio.conf \
+    --selector 'crio.image.pause_image' \
+    "${PAUSE_CONTAINER}"
+  rm --force /run/dockershim.sock
+  ln -sf /run/crio/crio.sock /run/dockershim.sock
 else
-    echo "Container runtime ${CONTAINER_RUNTIME} is not supported."
-    exit 1
+  echo "Custom container runtime."
 fi
 
 echo "Configuring kubelet snap"
